github_helper.py

```diff
@@ -0,0 +1,132 @@
+"""
+github_helper provides several methods to help interface with Github
+"""
+
+import requests
+
+GITHUB_ORG = 'apache'
+SECRET_FILE = '../../github_auth/apache-beam.secret'
+SOURCE_REMOTE = 'github'
+TARGET_BRANCH = 'master'
+TARGET_REMOTE = 'apache'
+
+GITHUB_API_ROOT = 'https://api.github.com'
+GITHUB_REPO_FMT_URL = GITHUB_API_ROOT + '/repos/{0}/{1}'
+
+BOT_NAME = 'merge-bot'
+GITHUB_SECRET = '../../github_auth/mergebot.secret'
+
+
+class GithubHelper:
+  """
+  GithubHelper provides a lightweight authenticated wrapper for the Github API.
+  """
+  def __init__(self, org, repo):
+    self.org = org
+    self.repo = repo
+    with open(GITHUB_SECRET, 'r') as key_file:
+      self.bot_key = key_file.read().strip()
+    print('Loaded key file.')
+
+  def fetch_prs(self):
+    """
+    Gets the PRs for the configured repository
+
+    :return: returns a JSON object containing the pull request if successful,
+             else None.
+    """
+    print('Loading pull requests from Github at {}.'.format(PULLS_URL))
+    r = requests.get(pulls_url(self.org, self.repo),
+                     auth=(BOT_NAME, self.bot_key))
+    if r.status_code != 200:
+      print('Oops, that didn\'t work. Error below,
+            waiting then trying again.')
+      print(r.text)
+      return None
+    print('Loaded.')
+    return r.json()
+
+  def fetch_comments(self, pr_num):
+    """
+    Gets the comments for a specified pull request
+
+    :return: returns a JSON object containing the comments if successful, else
+             None.
+    """
+    cmt_url = comment_url(self.org, self.repo, pr_num)
+    print('Loading comments from Github at {}.'.format(cmt_url))
+    r = requests.get(cmt_url, auth=(BOT_NAME, self.bot_key))
+    if r.status_code != 200:
+      print('Oops, that didn\'t work. Error below, moving on.')
+      print(r.text)
+      return None
+    return r.json()
+
+  def post_error(self, content, pr_num):
```

Comment:
For future proofing, and readability, it may make sense to return an object to interact with from the `fetch_prs` method. That object could have appropriate accessors, as well as methods such as `post_error` that don't require the pr_num (thus avoiding mistakes).